packages/lit-ssr/README.md

```diff
@@ -0,0 +1,158 @@
+# lit-ssr
+
+lit-html & LitElement SSR server
+
+## Setup
+
+This repo requires Node 14+ because of it's use of experimental VM modules for
+creating VM sandbox for executing the client-side code in. It also requires
+templates/elements be authored using pre-release versions of lit-html@2 and
+lit-element@3.
+
+## Usage
+
+### High-level API: render HTML pages
+
+The easiest way to get started using lit-ssr is to add the `renderHTMLFile`
+middleware from the `koa-render-html-file` module to your server. This
+middleware will server-side render any `Lit` custom elements included in html
+files served in your application. It handles URLs to files with `.html` or bare
+paths with `accept: text/html` or `accept: *`. If a static file does not exist
+at the specified URL path, the `fallback` file will be rendered/served.
+
+Below is an example usage of the lit-ssr `renderHTMLFile` middleware:
+
+```js
+// Example server.js
+
+import * as path from 'path';
+import Koa from 'koa';
+import staticFiles from 'koa-static';
+import koaNodeResolve from 'koa-node-resolve';
+import {renderHTMLFile} from 'lit-ssr/lib/koa-render-html-file.js';
+
+const {nodeResolve} = koaNodeResolve;
+
+const root = /* Root of your webapp */;
+
+(async() => {
+
+  const port = 8080;
+  const app = new Koa();
+  app.use(nodeResolve({root}));
+  app.use(renderHTMLFile({root, fallback: 'index.html'})); // Use `renderHTMLFile` plugin
+  app.use(staticFiles(root));
+  app.listen(port, () => {
+    console.log(`Server listening on port ${port}`);
+  });
+
+})();
+```
+
+When using this middleware, there are a few steps you should do to prepare you
+HTML files for proper server rendering:
+
+- Add the `template-shadowroot` polyfill to hydrate server-rendered `<template shadowroot>` elements. For example:
+  ```html
+  <script type="module">
+    import {hydrateShadowRoots} from './node_modules/template-shadowroot/template-shadowroot.js';
+    hydrateShadowRoots(document.body);
+  </script>
+  ```
+- Add `ssr` attribute to scripts that _should_ be executed on the server, for
+  example, scripts that include custom element definitions. For example:
+  ```html
+  <script type="module" src="my-app-components.js" ssr></script>
+  ```
+- (Optional): Add `type="ssr-only"` attribute to scripts that should _only_ be
```

Comment:
It's sorta nice to use `type` for things that should _only_ run on the server so that they can be left in the page when testing client-side only. That's why I was using both.